pkg/descheduler/policyconfig.go

```diff
@@ -163,10 +164,21 @@ func V1alpha1ToInternal(
 					},
 				}
 
-				// Plugins have either of the two extension points
-				if pluginbuilder.PluginRegistry[string(name)].Balance {
+				// Plugins have either of the two extension points right now
+				// this might change in the future
+				exampleArgs := pluginbuilder.PluginRegistry[string(name)].PluginArgInstance
+				pluginInstance, err := pluginbuilder.PluginRegistry[string(name)].PluginBuilder(exampleArgs, &handleImpl{})
+				if err != nil {
+					klog.ErrorS(fmt.Errorf("could not build plugin"), "plugin build error", "plugin", name)
+					return nil, fmt.Errorf("could not build plugin: %v", name)
+				}
+
+				switch p := pluginInstance.(type) {
```

Comment:
A plugin can implement both BalancePlugin and DeschedulePlugin interfaces. Right now there's no such plugin. Though, later we might introduce one and forget to refactor the lines.

Reply:
If we add another type this can become a mess of combinations. I have added inner switches then, let me know if another approach would be better

Reply:
separated them so we check each and both checks can happen for same plugin